src/main/java/com/fasterxml/classmate/ResolvedType.java

```diff
@@ -96,6 +96,26 @@ public final boolean canCreateSubtype(Class<?> subtype) {
      */
     public abstract List<ResolvedType> getImplementedInterfaces();
 
+    /**
+     * Returns Ordered list of parent, current and interfaces of this type.
+     *
+     * @return List of parent, current and interfaces of this type containing at least current type
+     */
+    public List<ResolvedType> getAllTypes() {
```

Comment:
Needs to resolve super types recursively, as methods called will only include immediate super class, interfaces implemented; but not super types of those types.

I think name should be `getAllSuperTypes()`; I also think that list should NOT include type itself (so that name makes sense in that context).
It might make sense to refactor this so that were two variants:

1. `getAllSuperTypes()` to work as described
2. `addAllSuperTypes(Collection<ResolvedType>)` that would add types into given set

... and come to think of this, I realized that `List` does not quite make sense here, for two reasons:

1. There is no well-defined ordering for inclusion; for this reason, we could just use `Collection`. But further,
2. Since it will be necessary to avoid duplicate interfaces (there are cases where things like `java.io.Serializable` are implemented via multiple super-interfaces).

So most likely return type really should be `Set`. This does open up some further questions (should differently parameterized super-types -- say, `Comparable<X>` vs `Comparable<Y>`; or, some cases, "raw" variant -- be included once or multiple times), but for now simple equality could work (which would lead to double inclusion for varying type parameters).

Finally, there may be a problem with recusive types (most common being `Enum`...), so a test for that might be needed. But I can take care of that.